pkg/server/commands/read_authzmodel.go

```diff
@@ -7,13 +7,13 @@ import (
 	openfgav1 "github.com/openfga/api/proto/openfga/v1"
 	"github.com/openfga/openfga/pkg/logger"
 	serverErrors "github.com/openfga/openfga/pkg/server/errors"
-	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/typesystem"
 )
 
 // ReadAuthorizationModelQuery retrieves a single type definition from a storage backend.
 type ReadAuthorizationModelQuery struct {
-	backend storage.AuthorizationModelReadBackend
-	logger  logger.Logger
+	logger             logger.Logger
+	typesystemResolver typesystem.TypesystemResolver
```

Comment:
If in the future this command needs to read something else aside from the model from the DB, we will have to undo this change and bring back the `backend` field.

This comment applies to every other command where we make this update.

Reply:
> If in the future this command needs to read something else aside from the model from the DB, we will have to undo this change and bring back the backend field.

This is a bottomless point. You could say the same thing of so many other design choices. Future proofing the decision based on what future things this may need to do with a database or not is irrelevant to the composition here at this time. The command for `ReadAuthorizationModel` needs to be able to resolve a TypeSystem. It doesn't care how that is done. It could be done via a database, a file, stdin, or some remote RPC. It doesn't care. What it needs is something that can resolve a TypeSystem, and the TypesystemResolver signature provides that functionality uniquely.